_posts/en/newsletters/2021-03-31-newsletter.md

```diff
@@ -0,0 +1,182 @@
+  can be better for privacy.
+
+## Releases and release candidates
+
+*New releases and release candidates for popular Bitcoin infrastructure
+projects.  Please consider upgrading to new releases or helping to test
+release candidates.*
+
+- [HWI 2.0.1][] is a bugfix release that addresses minor issues with Trezor
+  T passphrase entry and keyboard shortcuts in the `hwi-qt` user
+  interface.
+
+- [C-Lightning 0.10.0-rc1][C-Lightning 0.10.0] is a release candidate
+  for the next major version of this LN node software.
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo],
+[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
+[Rust-Lightning][rust-lightning repo], [libsecp256k1][libsecp256k1
+repo], [Hardware Wallet Interface (HWI)][hwi repo],
+[Rust Bitcoin][rust bitcoin repo], [BTCPay Server][btcpay server repo],
+[Bitcoin Improvement Proposals (BIPs)][bips repo], and [Lightning
+BOLTs][bolts repo].*
+
+- [Bitcoin Core #17227][] adds a new `make apk` target to the build
+  system which packages bitcoin-qt for the Android operating system. This
+  continues [previous work][news 72 android ndk] which added support for
+  packaging the Android NDK.  Also included are [documentation][android build
+  doc] for building Bitcoin Core for Android and a [continuous integration
+  job][android ci] to test the Android build system.
+
+- [Rust-Lightning #849][] makes a channel's `cltv_expiry_delta`
+  configurable and reduces the default value from 72 blocks to 36
+  blocks.  This parameter sets the minimum number of blocks a node will
+  have to settle a payment attempt with its upstream peer after learning
+  from its downstream peer whether or not that payment succeeded; it must
+  be long enough to settle the payment onchain if necessary but should
+  be short enough that it's competitive with other nodes that are trying
+  to minimize possible delays.  See also [Newsletter #40][news40
+  cltv_expiry_delta] where LND reduced its value to 40 blocks.
+
+- [C-Lightning #4427][] makes it possible to experiment with
+  dual funded payment channels by using the configuration option
+  `--experimental-dual-fund`.  Dual funding allows funds for the initial
+  channel balance to be contributed by both the node initiating the
+  channel and the node accepting the channel, which can be useful for
+  merchants and other users who want to begin receiving payments as
+  soon as the channel finishes opening.
+
+- [Eclair #1738][] corrects the penalizing of revoked HTLC transactions by
+  spending all outputs spendable by the revocation key instead of just the
+  first. This is to account for the possibility that [anchor outputs][topic
+  anchor outputs] enabled peers merge their
+  `SIGHASH_SINGLE|SIGHASH_ANYONECANPAY` HTLC transactions, resulting in an
+  aggregate transaction with multiple outputs to be penalized.
```

Comment:
What about:

```suggestion
- [Eclair #1738][] updates the penalty mechanism for [anchor outputs][topic anchor outputs]
  revoked HTLC transactions. Anchor outputs introduces the possibility
  to merge multiple `SIGHASH_SINGLE|SIGHASH_ANYONECANPAY` HTLC transactions,
  resulting in an aggregate transaction with multiple outputs to be penalized at once.
  The PR ensures that all outputs that are spendable with the revocation key are claimed
  instead of claiming only one per transaction.
```

Thanks again for highlighting this PR, it could be useful for other implementers who may not have noticed this edge case.

Reply:
Change looks good to me. Thanks @t-bast!

Reply:
Made a different edit that incorporates several of @t-bast's suggested changes (thanks!).  In past writing ([e.g.](https://bitcoinops.org/en/newsletters/2020/12/02/#proposed-intermediate-solution-for-ln-sighash-single-fee-theft)) I've tried not to refer to the sighash changes as anchor outputs, since it's independent of the changes made to use CPFP carve out---it just happened to have been made in the same protocol change as anchor outputs.  My edited version tries to make that distinction (and also doesn't backlink to this item from our anchor outputs topic for that reason) and also adds a link to a recent news item where we mentioned HTLC enforcement batching.

Thanks @dongcarl for the writeup and @t-bast for not only reviewing but also suggesting edits!